src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -353,6 +355,91 @@ public String addDatasetAssetSpecifications(
         .submit();
   }
 
+  public void validateDatasetAssetSpecification(Dataset dataset, AssetModel assetModel) {
+    // Validate Root Table
+    DatasetTable rootTable =
+        dataset.getTables().stream()
+            .filter(datasetTable -> datasetTable.getName().equals(assetModel.getRootTable()))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new InvalidAssetException(
+                        "Root table " + assetModel.getRootTable() + " does not exist in dataset."));
+    // Validate Root Column
+    if (!rootTable.getColumns().stream()
+        .map(c -> c.getName())
+        .toList()
+        .contains(assetModel.getRootColumn())) {
+      throw new InvalidAssetException(
+          "Root column "
+              + assetModel.getRootColumn()
+              + " does not exist in table "
+              + rootTable.getName());
+    }
+    // Validate Tables
+    assetModel
+        .getTables()
+        .forEach(
+            assetTable -> {
+              DatasetTable currentTable =
+                  dataset.getTables().stream()
+                      .filter(datasetTable -> datasetTable.getName().equals(assetTable.getName()))
+                      .findFirst()
+                      .orElseThrow(
+                          () ->
+                              new InvalidAssetException(
+                                  "Table " + assetTable.getName() + " does not exist in dataset."));
```

Comment:
There might not be a way to do it, but it would be nice if the (almost) duplicate code (with lines 360-378 above) could be combined.

Reply:
Done!